storage/mysql/log_storage.go

```diff
@@ -191,6 +206,9 @@ func (t *readOnlyLogTX) GetActiveLogIDsWithPendingWork(ctx context.Context) ([]i
 }
 
 func (m *mySQLLogStorage) beginInternal(ctx context.Context, treeID int64, readonly bool) (storage.LogTreeTX, error) {
+	once.Do(func() {
+		createMetrics(prometheus.MetricFactory{})
```

Comment:
Ideally, yes, although in practice I don't think we'll ever use the MySQL storage layer with anything other than the Prometheus metric layer.

I'll add a TODO